client/views/3-MyProfileView.js

```diff
@@ -0,0 +1,263 @@
+      else return Linking.openURL(url);
+    }).catch(err => console.error('An error occurred', err));
+  };
+
+  stringifyMyProfile = (myProfile) => {
+    const myProfileCompressed = compressProfile(myProfile);
+    const jsonString = JSON.stringify(myProfileCompressed);
+    this.props.storeMyProfileJSONString(jsonString);
+  };
+
+  componentDidMount () {
+    this.stringifyMyProfile(this.props.me);
+  };
+
+  render() {
+    const myProfile = this.props.me;
+    
+    const { fName, lName, occupation, currentLocation: loc, email, birthplace  } = myProfile.personal;
+    const { facebook, instagram, twitter, blog } = myProfile.social;
+    const { linkedIn, github, cv, website } = myProfile.networking;
+    const {photo} = myProfile;
+    return (
+      
+      <View style={styles.container}>
+        <View style={styles.contactsWrapper}>
+          <Avatar
+            xlarge
+            rounded
+            source={{ uri: photo }}
+            onPress={() => console.log("Works!")}
+          />
+          <Text style={styles.name}> {`${fName} ${lName}`} </Text>
+          <Text style={styles.jobtitle}> {`${occupation} at Codeworks`} </Text>
+          <Text style={styles.details}>
+            Lives in - {`${loc.country}, ${loc.place}`}
+          </Text>
+          <Text style={styles.category}>from {`${birthplace.country}`} </Text>
+
+          <TouchableOpacity
+            style={styles.buttonRound}
+          >
+            <Icon 
+              type='font-awesome'
+              name='share-alt' 
+              size={36} 
+              color="black"
+              onPress={() => this.props.navigation.navigate('ShareQRCode')}
+            />
+          </TouchableOpacity>
+
+        </View>
+        {/* <Text style={styles.category}> Social </Text>
+        <View style={styles.iconsContainer}>
+          <Icon
+            type='entypo'
+            name='facebook'
+            size={28}
+            raised={true}
+            onPress={() => this.canOpenURL(facebook)}
+          />
+
+          <Icon style={styles.icons}
+            type='font-awesome'
+            name='instagram'
+            color='#00aced'
+            size={28}
+            raised={true}
+            onPress={() => this.canOpenURL(instagram)}
+          />
+
+          <Icon style={styles.icons}
+            type='font-awesome'
+            name='twitter'
+            color='#517fa4'
+            size={28}
+            raised={true}
+            onPress={() => this.canOpenURL(twitter)}
+          />
+
+          <Icon style={styles.icons}
+            type='font-awesome'
+            name='meetup'
+            color='#517fa4'
+            size={28}
+            raised={true}
+            onPress={() => this.canOpenURL(blog)}
+          />
+        </View> */}
+
+        <Text style={styles.category}> Networking </Text>
+
+        <ScrollView horizontal contentContainerstyle={styles.iconsContainer}>
+          <Icon
```

Comment:
I think here you could just create and Icon component incase you need it later and then map of state:
{this.state.valuesToPassProps.map(value =>{
  <Icon
 type='value.entypo'
            name='value.linkedin'
            size={value.28}
            raised={value.boolean}
            onPress={value.linkedIn}
})